OpenMarket/OpenMarket/ViewController.swift

```diff
@@ -6,13 +6,229 @@
 
 import UIKit
 
+enum ArrangeMode: Int {
+    case list = 0
+    case grid = 1
+}
+
 class ViewController: UIViewController {
+    private let segmentedControl = UISegmentedControl(items: ["LIST", "GRID"])
+    private var arrangeMode: ArrangeMode = .list
+    private var data: [Product] = []
+    private lazy var collectionView = UICollectionView(frame: .zero, collectionViewLayout: listLayout())
+    private lazy var activityIndicator: UIActivityIndicatorView = {
+        createActivityIndicator()
+    }()
+}
 
+// MARK: - Life Cycle
+extension ViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        setUpNavigationItems()
+        self.view.addSubview(collectionView)
+        self.view.addSubview(activityIndicator)
+        self.activityIndicator.startAnimating()
+        receiveProductListData()
+        
+        setUpSegmentedControlLayout()
+        setUpCollectionViewConstraints()
+        collectionViewDelegate()
+        
+        setUpInitialState()
     }
+}
 
+// MARK: - Delegate
+extension ViewController {
+    func collectionViewDelegate() {
+        collectionView.delegate = self
+        collectionView.dataSource = self
+    }
+}
 
+// MARK: - Delegate Method
+extension ViewController: UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        data.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        switch self.arrangeMode {
+        case .list:
+            return configureListCell(indexPath: indexPath)
+        case .grid:
+            return configureGridCell(indexPath: indexPath)
+        }
+    }
+}
+
+// MARK: - Private Method
+extension ViewController {
```

Comment:
private 메서드를 extension으로 묶으시려면 모든 메서드에 private를 추가하지 않고 extension 자체에 private 키워드를 추가해주는건 어떨까요?
```suggestion
// MARK: - Private Method
private extension ViewController {
```

Reply:
private를 함수에 붙여주거나, extension 자체에 붙여주는 방법이 있다는 것을 알게되었습니다. extenstion 붙여주었습니다!